zephyr/lib/regions_mm.c

```diff
@@ -69,23 +69,24 @@ struct vmh_heap *vmh_init_heap(const struct vmh_heap_config *cfg,
 		goto fail;
 
 	/* If no config was specified we will use default one */
-	if (!cfg) {
+	if (!cfg)
 		vmh_get_default_heap_config(new_heap->virtual_region, &new_config);
-		cfg = &new_config;
-	}
+	else
+		memcpy_s(&new_config, sizeof(struct vmh_heap_config),
```

Comment:
`sizeof(new_config)`

Reply:
Old code resulted in the following behavior:
cfg was read correctly only in the if context and reads done after that returned garbage.
I do not understand the origin of the issue but sticking to new_config which was created and zeroed in the function body seems to fix the issue.

Reply:
> Old code resulted in the following behavior: cfg was read correctly only in the if context and reads done after that returned garbage. I do not understand the origin of the issue but sticking to new_config which was created and zeroed in the function body seems to fix the issue.

@dabekjakub sorry, I have a substantial issue with fixing something that we don't understand, especially if that's our own open-source software, not some 3rd-party black box.

Reply:
Well, I agree but it is either this or the code not working. Which do we choose? This can be investigated further but this fixes heap creation. 

Both ptr should be stack ptrs since cfg should be created runtime and not assigned space on the heap. At least that is an assumption I am making it is a client that will decide from where he will pass config.